posthog/models/filters/filter.py

```diff
@@ -111,15 +111,30 @@ def _parse_breakdown(self, data: Dict[str, Any]) -> Optional[Union[str, List[Uni
             return breakdown
 
     def _parse_target_entity(self, target_entity_data) -> Optional[Entity]:
-        if target_entity_data:
+        if isinstance(target_entity_data, dict):
+            return Entity(target_entity_data)
+        elif target_entity_data:
             data = json.loads(target_entity_data)
             return Entity({"id": data["id"], "type": data["type"]})
         return None
 
     def to_dict(self) -> Dict[str, Any]:
+
+        date_to = self.date_to or timezone.now()
+        if self.interval == "hour" or self.interval == "minute":
+            date_to = date_to
+        else:
+            date_to = date_to.replace(hour=0, minute=0, second=0, microsecond=0)
+
+        date_from = self.date_from or relative_date_parse("-11d")
+        if self.interval == "hour" or self.interval == "minute":
+            date_from = date_from
+        else:
+            date_from = date_from.replace(hour=0, minute=0, second=0, microsecond=0)
+
```

Comment:
not DRY. Going to change how this is done in follow-up PR that focuses on the filters